server/routes/personalRouter.ts

```diff
@@ -129,6 +173,15 @@ export default function personalRouter(services: Services): Router {
     submit: {
       audit: Page.PostEditWeight,
       method: personalController.weight().imperial.submit,
+      validation: {
+        validators: [weightImperialValidator],
+        onError: (req, res) => {
+          const { stone, pounds } = req.body
+          req.flash('stoneValue', stone)
+          req.flash('poundsValue', pounds)
+          return res.redirect(`/prisoner/${req.params.prisonerNumber}/personal/edit/weight/imperial`)
+        },
+      },
     },
   })
 
```

Comment:
I wonder if we need to be so specific with the flash messages. If we store `req.flash('inputValues', JSON.stringify(req.body)` then we don't need a specific onError and could just supply a redirect url (if we are unhappy to use the referrer). Then the GET routes would routinely look for flash.inputValues and parse it and the views just look for the same key as the input name. Would make it even less intrusive